ambry-account/src/main/java/com/github/ambry/account/MySqlAccountService.java

```diff
@@ -146,18 +177,43 @@ protected Account getAccountByNameNew(String accountName) {
     return cachedAccountServiceNew.getAccountByName(accountName);
   }
 
+  /**
+   * This method is used for uploading data to old db for testing purpose only.
+   */
+  protected Account getAccountByIdOld(short accountId) {
+    return cachedAccountService.getAccountById(accountId);
+  }
+
+  /**
+   * This method is used for uploading data to old db for testing purpose only.
+   */
+  protected Account getAccountByNameOld(String accountName) {
+    return cachedAccountService.getAccountByName(accountName);
+  }
+
   protected CachedAccountService getCachedAccountService() {return cachedAccountService;}
 
   @Override
   public void updateAccounts(Collection<Account> accounts) throws AccountServiceException {
     cachedAccountService.updateAccounts(accounts);
     if (config.enableNewDatabaseForMigration) {
-      cachedAccountServiceNew.updateAccounts(accounts);
+      try {
+        cachedAccountServiceNew.updateAccounts(accounts);
+      } catch (AccountServiceException e) {
+        if (config.ignoreNewDatabaseUploadError) {
```

Comment:
can we add a metric for all the ignored errors?

Reply:
Updated.